### PR TITLE
Add py.typed to package

### LINKED ISSUE #249: Add a py.typed (PEP-561)
https://www.python.org/dev/peps/pep-0561/

We're providing static typing, we should definitely expose it !

### PR DESCRIPTION
**Pull request by SBillion**
Closes #249 

### Successful PR Checklist:

- [ ] Tests
  - [x] (not applicable?)
- [ ] Documentation
  - [x] (not applicable?)
- [x] Had a good time contributing?
- [ ] (Maintainers: add PR labels)

### CODE REVIEW

**Comment by SBillion** [comment]
Had to set zip_safe to False or mypy won't find the installed package
https://mypy.readthedocs.io/en/latest/installed_packages.html?highlight=pep#making-pep-561-compatible-packages

File: setup.cfg [hunk 1]
```diff
@@ -72,6 +72,9 @@ include =
 console_scripts =
     procrastinate = procrastinate.cli:main
 
+[options.package_data]
+procrastinate = py.typed
```
Comment (ewjoachim) [inline setup.cfg:76]: Why do we need that with the MANIFEST.in ?

File: setup.cfg [hunk 2]
```diff
@@ -17,7 +17,7 @@ classifiers =
     License :: OSI Approved :: MIT License
 
 [options]
-zip_safe = True
+zip_safe = False
```
Comment (ewjoachim) [inline setup.cfg:20]: Would be nice to mention https://mypy.readthedocs.io/en/stable/installed_packages.html#making-pep-561-compatible-packages

File: setup.cfg (hunk 1 above)
Comment (SBillion) [reply setup.cfg]: Yes you're right. I added it to MANIFEST.in after and forgot to remove here. Done

File: setup.cfg (hunk 2 above)
Comment (SBillion) [reply setup.cfg:21]: Done

**Pull request merged**